File: src/SampleCRM/Views/BaseChildWindow.cs
using System.Windows;
using System.Windows.Controls;

namespace SampleCRM.Web.Views
{
    public class BaseChildWindow : ChildWindow
    {
        protected virtual double windowMobileSizeMult => 1d;
        protected virtual double windowSizeMult => .85d;
        protected virtual double MaxMobileWidth => 700d;

        public BaseUserControl InnerControl { get; protected set; }

        public bool IsMobileUI => Application.Current.MainWindow.ActualWidth <= MaxMobileWidth;

        public BaseChildWindow()
        {
            arrangeSize();
            Application.Current.MainWindow.SizeChanged += MainWindow_SizeChanged;
            Loaded += BaseChildWindow_Loaded;
            Unloaded += BaseChildWindow_Unloaded;
        }

        private void BaseChildWindow_Unloaded(object sender, RoutedEventArgs e)
        {
            if (AsyncHelper.ContentPage != null)
                AsyncHelper.ContentPage.MakeBlur(false);
        }

        private void BaseChildWindow_Loaded(object sender, RoutedEventArgs e)
        {
            if (AsyncHelper.ContentPage != null)
                AsyncHelper.ContentPage.MakeBlur(true);
        }

        private void MainWindow_SizeChanged(object sender, WindowSizeChangedEventArgs e)
        {
            arrangeSize();
        }

        private void arrangeSize()
        {
            if (IsMobileUI)
            {
                Width = Application.Current.MainWindow.ActualWidth * windowMobileSizeMult;
                Height = Application.Current.MainWindow.ActualHeight * windowMobileSizeMult;
            }
            else
            {
                Width = Application.Current.MainWindow.ActualWidth * windowSizeMult;
                Height = Application.Current.MainWindow.ActualHeight * windowSizeMult;
            }
        }
    }
}

File: src/SampleCRM/Views/BaseChildWindow.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace Views
    Public Class BaseChildWindow
        Inherits ChildWindow

        Protected Overridable ReadOnly Property windowMobileSizeMult As Double
            Get
                Return 1.0R
            End Get
        End Property

        Protected Overridable ReadOnly Property windowSizeMult As Double
            Get
                Return 0.85R
            End Get
        End Property

        Protected Overridable ReadOnly Property MaxMobileWidth As Double
            Get
                Return 700.0R
            End Get
        End Property

        Public Property InnerControl As BaseUserControl

        Public ReadOnly Property IsMobileUI As Boolean
            Get
                Return Application.Current.MainWindow.ActualWidth <= MaxMobileWidth
            End Get
        End Property

        Public Sub New()
            arrangeSize()
            AddHandler Application.Current.MainWindow.SizeChanged, AddressOf MainWindow_SizeChanged
        End Sub

        Private Sub MainWindow_SizeChanged(ByVal sender As Object, ByVal e As WindowSizeChangedEventArgs)
            arrangeSize()
        End Sub

        Private Sub arrangeSize()
            If IsMobileUI Then
                Width = Application.Current.MainWindow.ActualWidth * windowMobileSizeMult
                Height = Application.Current.MainWindow.ActualHeight * windowMobileSizeMult
            Else
                Width = Application.Current.MainWindow.ActualWidth * windowSizeMult
                Height = Application.Current.MainWindow.ActualHeight * windowSizeMult
            End If
        End Sub
    End Class
End Namespace


